presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/ArraySqlFunctions.java

```diff
@@ -136,4 +136,14 @@ public static String removeNulls()
     {
         return "RETURN IF(none_match(input, x -> x is null), input, filter(input, x -> x is not null))";
     }
+
+    @SqlInvokedScalarFunction(value = "array_least_frequent", deterministic = true, calledOnNullInput = true)
+    @Description("Determines the least frequent elemement in the array. If there are multiple elements, the function returns the biggest element")
+    @TypeParameter("T")
+    @SqlParameter(name = "input", type = "array(T)")
+    @SqlType("T")
+    public static String array_least_frequent()
+    {
+        return "RETURN IF(input IS NULL, NULL, IF(CARDINALITY(REMOVE_NULLS(input)) = 0, NULL, MAP_KEYS(MAP_TOP_N(TRANSFORM_VALUES(ARRAY_FREQUENCY(REMOVE_NULLS(input)), (k, v) -> -1 * v), 1))[1]))";
```

Comment:
two things 
 * see if you want to N as a param - n least (and most) frequent
 * also see if you can use map_entries, transform them to (v,k) and sort/reverse

Reply:
Hmm...we can take N as param to cover more generalised cases. Let me take N as optional parameter default to 1. WDYT?